src/Walker/Collector.C

```diff
@@ -113,33 +59,46 @@ Collector::chareOrdPDF( const std::vector< tk::UniPDF >& updf,
   for (const auto& p : tpdf) m_ordtpdf[i++].addPDF( p );
 
   // If all chares on my PE have contributed, send partial sums to host
-  if (m_nopdf == m_nchare) {
+  if (m_nord == m_nchare) {
+
+    // Create Charm++ callback function for reduction
+    CkCallback c1( CkReductionTarget( Distributor, estimateOrd ), m_hostproxy );
+
+    // Contribute partial sums to host via Charm++ reduction
+    contribute( static_cast< int >( m_ordinary.size() * sizeof(tk::real) ),
+                m_ordinary.data(), CkReduction::sum_double, c1 );
+
+    // Zero counters for next collection operation
+    std::fill( begin(m_ordinary), end(m_ordinary), 0.0 );
 
     // Serialize vector of PDFs to raw stream
     auto stream = tk::serialize( m_ordupdf, m_ordbpdf, m_ordtpdf );
 
     // Create Charm++ callback function for reduction.
     // Distributor::estimateOrdPDF() will be the final target of the reduction
     // where the results of the reduction will appear.
-    CkCallback cb( CkIndex_Distributor::estimateOrdPDF(nullptr), m_hostproxy );
+    CkCallback c2( CkIndex_Distributor::estimateOrdPDF(nullptr), m_hostproxy );
```

Comment:
"c2" is not a good variable name in any situation and will just make it harder for people to understand the code